src/System.Private.CoreLib/shared/System/IO/StreamWriter.cs

```diff
@@ -1135,6 +1072,15 @@ private bool HaveWrittenPreamble_Prop
                 await stream.FlushAsync(cancellationToken).ConfigureAwait(false);
             }
         }
-        #endregion
+
+        private void ThrowIfDisposed()
+        {
+            if (_disposed)
+            {
+                ThrowObjectDisposedException();
+            }
+
+            void ThrowObjectDisposedException() => throw new ObjectDisposedException(GetType().Name, SR.ObjectDisposed_WriterClosed);
```

Comment:
Just out of curiosity, what is the benefit of using this extra inline method?

Reply:
I wanted to avoid any potential issues around perf as part of this change, since I'm not validating perf impact.  The ThrowIfDisposed method is called in a bunch of places, and should be inlined, and I wanted to make sure that a) the extra IL associated with exception creation and throw didn't prevent that, and b) that even if it was inlined the extra asm associated SR access, the exception allocation, and the throw didn't get stamped into every call site.

Reply:
I see, so basically what this will cause, is that only the if check and the call to ThrowObjectDisposedException are inlined into the callsites of ThrowIfDisposed?

Reply:
Correct.

Reply:
Thanks for explaining 😄